ml-agents/mlagents/trainers/trainer/rl_trainer.py

```diff
@@ -142,16 +146,18 @@ def advance(self) -> None:
                 # being emptied, the trajectories in the queue are on-policy.
                 for _ in range(traj_queue.maxlen):
                     try:
-                        t = traj_queue.get_nowait()
+                        t = traj_queue.get(block=not empty_queue, timeout=0.05)
                         self._process_trajectory(t)
```

Comment:
should we still process the trajectory if we have exceeded our buffer size?

Reply:
That's a possibility - I guess the question is whether it is better to dump trajectories or exceed the buffer size. I'm leaning towards exceeding the buffer size. @andrewcoh do you have any preferences?

Ultimately we should change this loop so that we stop getting from the queue if we need to update.